openssl.sh

```diff
@@ -4,19 +4,32 @@ tag: OpenSSL_1_0_2o
 source: https://github.com/openssl/openssl
 prefer_system: (?!slc5|slc6)
 prefer_system_check: |
-  if [ `uname` = Darwin ]; then test -d `brew --prefix openssl || echo /dev/nope` || exit 1; fi; echo '#include <openssl/bio.h>' | c++ -x c++ - -I`brew --prefix openssl`/include -c -o /dev/null || exit 1; echo -e "#include <openssl/opensslv.h>\n#if OPENSSL_VERSION_NUMBER >= 0x10100000L\n#error \"System's GCC cannot be used: we need OpenSSL 1.0.x to build XrootD. We are going to compile our own version.\"\n#endif\nint main() { }" | cc -x c++ - -I`brew --prefix openssl`/include -c -o /dev/null || exit 1
+  # Notice that we cannot use brew anymore, because they moved to 1.1.x.
+  false
 build_requires:
- - zlib
+ - zlib:(?!osx)
  - "GCC-Toolchain:(?!osx)"
 ---
 #!/bin/bash -e
 
 rsync -av --delete --exclude="**/.git" $SOURCEDIR/ .
+case $ARCHITECTURE in 
+  osx*) 
+    EXTRA_DEPS="no-zlib enable-eec_nistp_64_gcc_128 no-comp"
+    EXTRA_OPTS="-Wno-nonportable-include-path"
+  ;;
+  *) 
+    EXTRA_DEPS=zlib
+    EXTRA_OPTS="-Wa,--noexecstack" 
+  :;
+esac
 
-./config --prefix="$INSTALLROOT"                   \
+export KERNEL_BITS=64
```

Comment:
@mfasDa allegedly this is the trick to make it build 64 bit on macOS.

Reply:
Works, but it results in an error in the install part:
```
DEBUG:Readout:OpenSSL:aliceo2: + install_name_tool -id /Users/markusfasel/alice/sw/osx_x86-64/OpenSSL/v1.0.2o-1/lib/libcrypto.1.0.0.dylib osx_x86-64/OpenSSL/v1.0.2o-1/lib/libcrypto.1.0.0.dylib
DEBUG:Readout:OpenSSL:aliceo2: error: /Library/Developer/CommandLineTools/usr/bin/install_name_tool: can't open input file: osx_x86-64/OpenSSL/v1.0.2o-1/lib/libcrypto.1.0.0.dylib for writing (Permission denied)
DEBUG:Readout:OpenSSL:aliceo2: error: /Library/Developer/CommandLineTools/usr/bin/install_name_tool: can't lseek to offset: 0 in file: osx_x86-64/OpenSSL/v1.0.2o-1/lib/libcrypto.1.0.0.dylib for writing (Bad file descriptor)
DEBUG:Readout:OpenSSL:aliceo2: error: /Library/Developer/CommandLineTools/usr/bin/install_name_tool: can't write new headers in file: osx_x86-64/OpenSSL/v1.0.2o-1/lib/libcrypto.1.0.0.dylib (Bad file descriptor)
DEBUG:Readout:OpenSSL:aliceo2: error: /Library/Developer/CommandLineTools/usr/bin/install_name_tool: can't close written on input file: osx_x86-64/OpenSSL/v1.0.2o-1/lib/libcrypto.1.0.0.dylib (Bad file descriptor)
```

Reply:
Do you also have:

https://github.com/alisw/alidist/pull/1906/files#diff-bc6a953101d106eb0b4914f811bcbb84R55-R57

?

Reply:
Works for OpenSSL! Thanks! Now of course the packages depending on OpenSLL run into errors and need to be adapted until the PR is merged.